src/ToDoList.tsx

```diff
@@ -1,7 +1,21 @@
 import React from 'react';
+import {useState} from "react";
+import Modal from '@mui/material/Modal';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import TextField from '@mui/material/TextField';
 import './ToDoList.css';
 
 export const ToDoList = () => {
+    const [openWindow, setOpenWindow] = useState(false);
+
+    const addNewTaskButton = () => {
+        if (openWindow) {
+            setOpenWindow(false);
+        } else {
+            setOpenWindow(true);
+        }
+    }
```

Comment:
```suggestion
    const addNewTaskButton = () => {
        setOpenWindow(!openWindow);
    }
```

Reply:
Also `addNewTaskButton ` seems like a bit weird name
Possible alternatives:

- just `onClick`
- `onAddNewTaskClick`
- `toggleAddNewTaskModal`

Reply:
Also you should use `useCallback` here